pkg/manager/internal.go

```diff
@@ -135,14 +136,18 @@ func (cm *controllerManager) Add(r Runnable) error {
 		return err
 	}
 
+	shouldStart := false
+
 	// Add the runnable to the leader election or the non-leaderelection list
 	if leRunnable, ok := r.(LeaderElectionRunnable); ok && !leRunnable.NeedLeaderElection() {
+		shouldStart = cm.started
 		cm.nonLeaderElectionRunnables = append(cm.nonLeaderElectionRunnables, r)
 	} else {
+		shouldStart = cm.started && (cm.isLeader || cm.resourceLock == nil)
```

Comment:
If you push take the suggestion from the other comment, this could be simplified to `shouldStart = cm.startedLeader`.